### PR TITLE
Move connection creation into scheduler.

### PR DESCRIPTION
**Pull request by dessalines**
- #3076

Hopefully should avoid the deadlocks by creating a new connection at the immediate use time.

### CODE REVIEW

File: src/scheduled_tasks.rs [hunk 1]
```diff
@@ -32,9 +32,6 @@ pub fn setup(db_url: String, user_agent: String) -> Result<(), LemmyError> {
   let mut scheduler = Scheduler::new();
 
   let mut conn_1 = PgConnection::establish(&db_url).expect("could not establish connection");
```
Comment (Nutomic) [inline src/scheduled_tasks.rs:34]: It seems like this connection will be kept open forever without being used. Better to move it and the functions which use it into a code block or a separate function.

File: src/scheduled_tasks.rs [hunk 2]
```diff
@@ -43,23 +40,31 @@ pub fn setup(db_url: String, user_agent: String) -> Result<(), LemmyError> {
   clear_old_activities(&mut conn_1);
 
   // Update active counts every hour
+  let url = db_url.to_owned();
   scheduler.every(CTimeUnits::hour(1)).run(move || {
-    active_counts(&mut conn_1);
-    update_banned_when_expired(&mut conn_1);
+    let mut conn_2 = PgConnection::establish(&url).expect("could not establish connection");
```
Comment (Nutomic) [inline src/scheduled_tasks.rs:45]: No need to number the variables, just call it `conn`.

**Comment by Nutomic** [comment]
Clippy is failing. The change looks good, though Im curious how reuse of connections could result in deadlocks.

**Comment by phiresky** [comment]
This won't affect deadlocks because deadlocks can only happen while a locking transaction or query is running, the amount of open connections isn't relevant.

Still might be a good idea to not keep the connections open though because each connection in PG takes up a process and goes against the connection limit.

**Review by Nutomic — APPROVED** [review]

**Pull request merged**